app/scripts/controllers/formAdd.js

```diff
@@ -0,0 +1,207 @@
+'use strict'
+
+/**
+ * @ngdoc function
+ * @name endslaverynowApp.controller:FormAddCtrl
+ * @description
+ * # FormAddCtrl
+ * Controller of the endslaverynowApp
+ */
+angular.module('endslaverynowApp')
+  .controller('FormAddCtrl',[
+    '$firebaseObject',
+    '$stateParams',
+    '$scope',
+    '$window',
+    'Upload',
+    function ($firebaseObject, $stateParams, $scope, $window, Upload) {
+      $scope.brandId = $stateParams.id
+      $scope.loaded = false
+
+      $scope.formType = ''
+      $scope.errorMessage = false
+
+      var itemTypes = {
+        'categories': {
+          name: 'Category',
+          getIdFunction: getCategoryId,
+          requiredInputs: {
+            name: true,
+            description: true,
+            purchaseUrl: false,
+            image: true,
+            category: false,
+            categories: false,
+            brand: false,
+            ranking: false,
+            parentCategoryId: false
+          },
+        },
+        'brands': {
+          name: 'Brand',
+          getIdFunction: getBrandId,
+          requiredInputs: {
+            name: true,
+            description: true,
+            purchaseUrl: false,
+            image: true,
+            category: false,
+            categories: true,
+            brand: false,
+            ranking: true
+          },
+        },
+        'products': {
+          name: 'Product',
+          getIdFunction: getProductId,
+          requiredInputs: {
+            name: true,
+            description: true,
+            purchaseUrl: true,
+            image: true,
+            categoryId: true,
+            brandId: true,
+            ranking: false,
+            parentCategoryId: false
+          },
+        }
+      }
+
+      $scope.rankingOptions = {
+        'good': 'Good',
+        'better': 'Better',
+        'best': 'Best'
+      }
+
+      var ref = firebase.database().ref()
+      var syncObject = $firebaseObject(ref)
+
+      $scope.products = []
+
+      syncObject.$loaded().then(function() {
+        $scope.brands = alphabetizeCollection(syncObject.brands)
+        $scope.categories = alphabetizeCollection(syncObject.categories.filter(Boolean))
+        $scope.products = syncObject.products || []
+        $scope.loaded = true
+      })
+
+      $scope.processForm = function(item) {
+        $scope.errorMessage = false
+        if(!item) {
+          $scope.errorMessage = true
+          return
+        }
+
+        var id = itemTypes[$scope.formType].getIdFunction()
+        item.id = id
+
+        if ($scope.formType === 'categories') {
```

Comment:
At some point, we should consider breaking out of using `formType` to determine what type of form we're inputting. For now though, this is fine